qiskit/ignis/experiments/calibration/methods/basic_1q.py

```diff
@@ -151,3 +155,88 @@ def __init__(self,
                          analysis=analysis,
                          job=job,
                          workflow=data_processing)
+
+
+class SinglePulseSingleParameterGenerator(Generator):
+    """
+    A generator that generates single pulses and scans a single parameter
+    of that pulse. Note that the same pulse may be applied to multiple qubits
+    to perform simultaneous calibration.
+    """
+
+    def __init__(self,
+                 qubits: Union[int, List[int]],
+                 parameters: Dict,
+                 amplitudes: List,
+                 scanned_parameter: str,
+                 pulse: Callable = None):
+        """
+        Args:
+            qubits: List of qubits to which this calibration can be applied.
+            parameters: The arguments for the pulse schedule.
+            amplitudes: A list of amplitudes for which to generate a circuit.
+            scanned_parameter: Name of the scanned parameter. If it is not present
+                in the parameters dictionary as a ParameterExpression it will
+                be added to this dict.
+            pulse: A parametric pulse function used to generate the schedule
+                that is added to the circuits. This defaults to Gaussian and
+                parameters should contain 'duration', and 'sigma'.
+        """
+        super().__init__('Single pulse generator', qubits)
+        self._pulse = pulse
+        self.parameters = parameters
+        self.amplitudes = amplitudes
+        self.scanned_parameter = scanned_parameter
+
+        # Add the parameter to be scanned if not supplied.
+        if scanned_parameter not in self.parameters:
+            self.parameters[scanned_parameter] = Parameter('α')
+        elif not isinstance(self.parameters[scanned_parameter], ParameterExpression):
+            self.parameters[scanned_parameter] = Parameter('α')
+
+        # Define the QuantumCircuit that this generator will use.
+        self.qc = QuantumCircuit(max(self._qubits)+1, max(self._qubits)+1)
```

Comment:
this should be always nqubits=1 because this is single qubit cal and agnostic to backend. qubit index is passed to experiment class and transpiler automatically maps to the correct physical qubit index with `initial_layout`.